src/v4/hooks/dca/DCAHook.sol

```diff
@@ -0,0 +1,479 @@
+    /// @param intent The decoded DCA intent
+    /// @param intentId The computed intent identifier
+    /// @param privateIntentHash The hash of private intent data
+    /// @param swapperSignature The swapper's signature
+    /// @param cosignerData The cosigner authorization data
+    /// @param cosignerSignature The cosigner's signature
+    /// @param resolvedOrder The resolved order to validate against
+    function _validateDCAIntent(
+        DCAIntent memory intent,
+        bytes32 intentId,
+        bytes32 privateIntentHash,
+        bytes memory swapperSignature,
+        DCAOrderCosignerData memory cosignerData,
+        bytes memory cosignerSignature,
+        ResolvedOrder calldata resolvedOrder
+    ) internal view {
+        // 1) Verify swapper signature (EIP-712) over full intent with privateIntentHash
+        _validateSwapperSignature(intent, privateIntentHash, swapperSignature);
+
+        // 2) Static field checks (binding correctness)
+        _validateStaticFields(intent, resolvedOrder);
+
+        // 3) Validate allocation structure (sum to 100%, no zeros)
+        _validateAllocationStructure(intent.outputAllocations);
+
+        // 4) Verify cosigner authorization
+        _validateCosignerSignature(intent, cosignerData, cosignerSignature);
+
+        // 5) State checks and period gating
+        _validateStateAndTiming(intentId, intent, cosignerData);
+
+        // 6) Chunk size checks
+        _validateChunkSize(intent, cosignerData, resolvedOrder.input.amount);
+
+        // 7) Price floor check (1e18 scaling)
+        _validatePriceFloor(intent, cosignerData);
+
+        // 8) Validate outputs match allocations and meet requirements
+        _validateOutputDistribution(intent, cosignerData, resolvedOrder.outputs);
+    }
+
+    function _transferInputTokens(ResolvedOrder calldata order, address to) private {
+        // TODO: Implement token transfer logic with AllowanceTransfer
+    }
+
+    /// @inheritdoc IDCAHook
+    function cancelIntents(uint256[] calldata nonces) external override {
+        for (uint256 i = 0; i < nonces.length; i++) {
+            _cancelIntent(msg.sender, nonces[i]);
+        }
+    }
+
+    /// @inheritdoc IDCAHook
+    function cancelIntent(uint256 nonce) external override {
+        _cancelIntent(msg.sender, nonce);
+    }
+
+    function _cancelIntent(address swapper, uint256 nonce) internal {
+        bytes32 intentId = keccak256(abi.encodePacked(swapper, nonce));
+        if (executionStates[intentId].cancelled) {
+            revert IntentAlreadyCancelled(intentId);
+        }
+        executionStates[intentId].cancelled = true;
+        emit IntentCancelled(intentId, swapper);
+    }
+
+    /// @notice Validates the swapper's EIP-712 signature over the DCA intent
+    /// @dev Verifies the signature using the provided private intent hash to reconstruct the full intent hash
+    /// @param intent The DCA intent to validate (with zeroed privateIntent field)
```

Comment:
not sure where is the best place to put this, but explaining why zeroing out the private intent will be nice